runtime/src/accounts_db.rs

```diff
@@ -1938,15 +1922,20 @@ impl AccountsDb {
         let mut store_accounts_timing = StoreAccountsTiming::default();
         if aligned_total > 0 {
             let mut start = Measure::start("find_alive_elapsed");
-            let mut accounts = Vec::with_capacity(alive_accounts.len());
+            let mut stored_accounts = Vec::with_capacity(alive_accounts.len());
             let mut hashes = Vec::with_capacity(alive_accounts.len());
             let mut write_versions = Vec::with_capacity(alive_accounts.len());
 
-            for (pubkey, alive_account) in alive_accounts {
-                accounts.push((pubkey, &alive_account.account));
-                hashes.push(alive_account.account_hash);
-                write_versions.push(alive_account.write_version);
+            for (_pubkey, alive_account) in alive_accounts.iter() {
+                stored_accounts.push(alive_account.account.clone_account());
```

Comment:
Cool, out of curiosity, what's the benefit of delaying the clone until here? Maybe some future change where we'll need a `StoredAccountMeta` early on instead of an `AccountSharedData`?

Reply:
If we find a duplicate of an account, then we only clone the last one instead of all of them. Shortly, we will not clone the account at all - we will implement ReadableAccount on StoredAccountMeta. This prevents us from having to make copies of anything, including the data.